device/device.yaml

```diff
@@ -6,33 +6,25 @@ info:
 
 components:
   schemas:
-    Device.Device:
+    Device:
       description: >-
-        An abstract container for emulated devices.
+        An abstract container for emulated protocol devices.
       type: object
-      required: [name]
+      required: [name, choice]
       properties:
         name: '../common/common.yaml#/components/x-inline/UniqueName'
-        devices_per_port: 
+        choice:
           description: >-
-            The number of emulated devices that will be created on each port.
-          type: integer
-          default: 1
-        ethernets:
-          description: >-
-            An unordered list of 0..n emulated ethernet protocols.
-          type: array
-          items:
-            $ref: './ethernet.yaml#/components/schemas/Device.Ethernet'
-        devices:
-          description: >-
-            An unordered list of 0..n emulated device containers.
-          type: array
-          items:
-            $ref: '#/components/schemas/Device.Device'        
-        networks:
-          description: >-
-            An unordered list of 0..n emulated device containers.
-          type: array
-          items:
-            $ref: './network.yaml#/components/schemas/Device.Network'        
+            The type of emulated protocol device. 
+          type: string
+          enum: [ethernet, vlan, ipv4, bgpv4]
```

Comment:
At this time I am holding off on adding bidirectional and any kind of meshing as this can be accomplished by adding more flows. If this proves to be a problem an separate issue can be opened.